src/App.php

```
@@ -26,13 +29,24 @@ public function __construct(ContainerInterface $container = null)
     public function run()
     {
         try {
-            $response = $this->container->get(RouteCollection::class)
-                ->dispatch(
-                    $this->container->get(ServerRequestInterface::class),
-                    new Response()
-            );
-        } catch (AssertionFailedException | InvalidRequestException $e) {
+            /** @var ServerRequestInterface $request */
+            $request = $this->container->get(ServerRequestInterface::class);
+            $response = new Response();
+            $response = $this->container->get(RouteCollection::class)->dispatch($request, $response);
+        } catch (AssertionFailedException $e) {
             $response = new BadRequestResponse([$e->getMessage()]);
+        } catch (NotFoundException $e) {
+            $response = new NotFoundResponse();
+        }  catch (MethodNotAllowedException $e){
+            if ($request->getMethod() !== 'OPTIONS'){
+                throw $e;
```

Do we want to throw this exception or Response with a 405?